Reorganize compliance-related docs pages (#2439)
chainguard-dev/edu: content/compliance/slsa/slsa-chainguard.md

---

> title: "SLSA at Chainguard"

Reviewer: ```suggestion
title: "SLSA at Compliance Chainguard"
```

aligning w/ H1 below

linktitle: "SLSA 1.1 Level 3 compliance at Chainguard"
description: "A brief overview of SLSA and Chainguard's compliance efforts."
type: "article"
date: 2025-07-23T01:24:23+00:00
lastmod: 2025-07-23T01:24:23+00:00
contributors: []
draft: false
tags: ["Compliance", "SLSA", "Standards"]
images: []
menu:
  docs:
    parent: "slsa"
weight: 020
toc: true
---

> # SLSA Compliance at Chainguard

Reviewer: ```suggestion
```

> SLSA (pronounced “salsa”), or Supply chain Levels for Software Artifacts, is a security framework consisting of standards and controls that prevent tampering, improve integrity, and secure packages and infrastructure. It is described in depth in [What is SLSA?](/open-source/slsa/what-is-slsa/).

Reviewer: ```suggestion
Supply chain Levels for Software Artifacts, or SLSA (pronounced "salsa"), is a security framework consisting of standards and controls that prevent tampering, improve integrity, and secure packages and infrastructure. It is described in depth in our overview on [What is SLSA?](/open-source/slsa/what-is-slsa/).
```

3 things here: 
1. I reversed the order of the name and the acronym, that's probably more a matter of preference than anything, so feel free to ignore
2. I think you may have smart quotes turned on wherever you drafted this (the quotes around `"salsa"` looked slanted to me). Not a huge deal but it's best to turn those off to avoid issues, especially in commands
3. The URL isn't working in the deploy preview which is odd. I think it's Lisa's code change that adds a `/` to urls. It looks right to me, but just double check it when you publish.

Author: 2: Oh, shoot. I copied that bit from the nascent blog post that is still being drafted in a gdoc. I usually catch those. Thanks!

Chainguard Containers and Guarded VMs are SLSA Level 3 compliant to provide confidence in the security of these products. Other Chainguard products, such as Chainguard Libraries, are SLSA Level 2 compliant and work is being done to bring them to level 3.

This page describes what we have done and are doing to bring Chainguard products into full SLSA Level 3 compliance.

## Isolated and Tamper-Resistent Builds

> Every Chainguard build now runs in a dedicated, ephemeral environment that starts from a clean slate. This build isolation ensures that no build can influence another, eliminating insider meddling or cross-contamination between builds. By provisioning short-lived build environments on demand, we greatly reduce the risk of any malware persistence or “cache poisoning” across builds.

Reviewer: can you define 'cache poisoning' or link to a resource that does?

## Unforgeable Provenance and Chain of Trust

> SLSA 3 requires that each artifact’s provenance (the record of how, when, and by whom it was built) be generated by a trusted service and cryptographically signed. In our process, the build system’s control plane creates this tamper-proof provenance data – build runners cannot falsify it. This means every container image or VM has an auditable “paper trail” back to its source, establishing a robust chain-of-trust for our software.

Reviewer: ```suggestion
SLSA 3 requires that each artifact’s provenance (the record of how, when, and by whom it was built) be generated by a trusted service and cryptographically signed. In our process, the build system’s control plane creates this tamper-proof provenance data — build runners cannot falsify it. This means every container image or VM has an auditable "paper trail" back to its source, establishing a robust chain-of-trust for our software.
```

Reviewer: My superpower is spotting en dashes that ought to be em dashes

## Secured Signing Keys and Secrets

We’ve further hardened our build platform to protect sensitive credentials. Signing keys and other secrets never reside on the build workers; instead, signing happens in a separate, secure service after the build completes. This separation ensures that even if a build job were compromised, it cannot access the keys needed to sign artifacts or attestations. By keeping signing operations isolated and secured, we prevent attackers or rogue insiders from forging trusted artifacts.

## Full Traceability and Transparency

> With SLSA 3, every Chainguard artifact comes with detailed metadata and documentation about how it was built. We generate build provenance and Software Bill of Materials (SBOM) for each release, with complete visibility into the software supply chain behind our Container and VM images.

Reviewer: ```suggestion
With SLSA 3, every Chainguard artifact comes with detailed metadata and documentation about how it was built. We generate build provenance and a Software Bill of Materials (SBOM) for each release, with complete visibility into the software supply chain behind our Container and VM images.
```

## Looking to the Future

> Reaching SLSA Level 3 compliance with any product is a major achievement. But, as Chainguard is a software security company, more is planned.

Reviewer: ```suggestion
Reaching SLSA Level 3 compliance with any product is a major achievement, but even more is planned.
```

> See [blog post] for more details.

Reviewer: missing hyperlink

Author: Yeah, waiting for it to post on 7/31. Thanks for the reminder!!

## Sample Provenance Code

> To pull an attestation, you must first [authenticate to Chainguard's registry](https://edu.chainguard.dev/chainguard/chainguard-images/chainguard-registry/authenticating/).

Reviewer: ```suggestion
To pull an attestation, you must first [authenticate to Chainguard's registry](/chainguard/chainguard-images/chainguard-registry/authenticating/).
```

Then you can download an attestation of provenance.

```
cosign download attestation \
  --predicate-type=https://slsa.dev/provenance/v1 \
  cgr.dev/chainguard/wolfi-base | jq -r .payload | base64 -d | jq .predicate
```

> To acquire this example:

Reviewer: ```suggestion
This example returns the following output:
```
